compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/types/util/BallerinaLangLibMethod.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.compiler.api.impl.types.util;
+
+import io.ballerina.compiler.api.symbols.Qualifier;
+import io.ballerina.compiler.api.types.FunctionTypeDescriptor;
+import io.ballerina.compiler.api.types.util.LangLibMethod;
+
+import java.util.Set;
+
+/**
+ * Represents a lang library function which can be called using a method call expr.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaLangLibMethod implements LangLibMethod {
```

Comment:
This looks identical to the `BallerinaMethodDeclaration`. Can we use the same thing to represent langlib methods as well?

Reply:
Yes, identical. Added a separate one because then one would be able to assign lang lib functions to `MethodDeclaration` vars and felt that that'll sorta sends the wrong message.